checker/AnDO_engine.py

```diff
@@ -18,77 +19,93 @@
 
 def parse_all_path(nested_list_of_dir):
     """
-    Transform this 
+    Transform this
     [
         ['Landing', 'sub-anye', '180116_001_m_anye_land-001', 'source'],
         ['Landing', 'sub-enya', '180116_001_m_enya_land-001', 'source'],
-        ['Landing', 'sub-enyo'], 
+        ['Landing', 'sub-enyo'],
         ['Landing', 'sub-enyo', '180116_001_m_enyo_land-001']
     ]
-    to 
+    to
     [
         ['Landing', 'sub-anye', '180116_001_m_anye_land-001', 'source'],
         ['Landing', 'sub-enya', '180116_001_m_enya_land-001', 'source'],
     ]
     Checking for the longest chain with the same sub chain
     """
-    
-    main_list = sorted(nested_list_of_dir, key= lambda sublist: len(sublist)) 
- 
-    # TODO : optimize 
-    i=0
-    j=1
-    while i < len(main_list) -1:
-        if j <= len(main_list) -1:
-            if len(main_list[i]) <= len(main_list[j]):
-                all_in = True
-                for elmt in main_list[i]:
-                    if elmt not in main_list[j]:
-                        all_in = False
-
-                if all_in:
-                    for elmt_to_add in main_list[j]:
-                        if elmt_to_add not in main_list[i]:
-                            main_list[i].append(elmt_to_add)
-                    main_list.pop(j)
-
-                else:
-                    if j < len(main_list) - 1:
-                        j+=1
-                    else:
-                        i +=1
-                        j =i+1
-            else:
-                i+=1
-                j=i+1
-        else:
-            break
+
+    def _test_is_included(my_list_of_lists, list_elem):
+        for my_list_elem in my_list_of_lists:
+            if all([val[0]==val[1] for val in zip(my_list_elem, list_elem)]):
+                return True
+
+        return False
+
+    new_list_of_lists = []
+
+    for list_elem in sorted(nested_list_of_dir,
+                            key= lambda sublist: len(sublist),
+                            reverse=True):
+
+        if not _test_is_included(new_list_of_lists, list_elem):
+            new_list_of_lists.append(list_elem)
+
+    return new_list_of_lists
+
+    #main_list = sorted(nested_list_of_dir, key= lambda sublist: len(sublist))
```

Comment:
lots of things commented here and below, is this normal?

Reply:
old version of the function should be deleted .

Reply:
did you delete it? if not, please do...

Reply:
Yes it has been deleted